[Instant Answer] Pokemon info

AkA84 opened the pull request:
**What does your Instant Answer do?**
It provides basic information and stats about a given pokemon


**What problem does your Instant Answer solve (Why is it better than organic links)?**
At a glance the user gets:
* A picture of the pokemon
* A description (randomly selected from the set given by the API)
* The types and egg groups the pokemon belongs to
* The name of the pokemon's next evolution (with a link to its IA)
* The basic stats

I think that's a good amount of information that the user gets before he/she even clicks any of the search results


**What is the data source for your Instant Answer? (Provide a link if possible)**
http://pokeapi.co/

According to the API author, API V1 should get deprecated this year and replaced by V2 ([source](http://phalt.co/if-you-have-data-they-will-consume-it/)). "Deprecated" in this case just means that the V1 will be rendered stable and it won't receive any more updates. I haven't found any information about the new API, but nothing prevents the IA to use V1 for now and then upgrade to V2 once it's out


**Why did you choose this data source?**
It's a nice, RESTful, widely used API


**Are there any other alternative (better) data sources?**
I kinda quickly settled on pokeapi since it seemed pretty good, but as far as I can tell I haven't found a decent competitor for it


**What are some example queries that trigger this Instant Answer?**
*pokemon pikachu*, *bulbasaur pokemon*


**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Pokemon lovers!


**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
https://duck.co/ideas/idea/4921/pok-mon-info


**Which existing Instant Answers will this one supersede/overlap with?**
None, as far as I know


**Are you having any problems? Do you need our help with anything?**
* I know that there is a built-in `record` template, but I couldn't figure out how to have it render a non-escaped string, which is necessary to have the link to the pokemon's evolution (i know about `Handlebars.SafeString` method but it doesn't seem to work as I suspect it's not something that the template expect to receive). I thought it would be cool thing to have since it allows the user to jumps back and forth different pokemon's evolutions, so for now I recreated the `record` mark-up while waiting to get some feedback from you
* On mobile the "auxiliary context" box disappears for my IA, but not for others. I noticed that the `.cw` wrapper is supposed to also get an `.has-aux` class to prevent it, but I'm not sure how to add that


**Where did you hear about DuckDuckHack? (For first time contributors)**
It popped up in conversation about privacy issues, and I found it quite interesting. It wasn't though until recently that I found out about your IA, and I'm really fascinated by the concept


**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
![pokemon-ia-example](https://cloud.githubusercontent.com/assets/6400898/6216192/d36d5354-b607-11e4-918c-3246356e0ccd.png)


## Checklist
Please place an 'X' where appropriate.

```
[x] Added metadata and attribution information
[x] Wrote test file and added to t/ directory
[x] Verified that Instant Answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[x] Verified that Instant Answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 10

    - Windows 7
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 8
        [x] IE 9
        [] IE 10

    - Windows XP
        [] IE 7
        [] IE 8

    - Mac OSX
        [x] Google Chrome
        [x] Firefox
        [] Opera
        [x] Safari

    - iOS (iPhone)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - iOS (iPad)
        [x] Safari Mobile
        [x] Google Chrome
        [] Opera

    - Android
        [x] Firefox
        [x] Native Browser
        [x] Google Chrome
        [] Opera
```

zekiel commented:
Thanks @AkA84!  Can you please fill out the pull request template and drop it in this thread? https://github.com/duckduckgo/duckduckgo-documentation/blob/master/templates/pull_request_template.md

AkA84 commented:
Sure, will do!

AkA84 commented:
Ok added, now going through your code styleguide and amending bits here and there of the code to be sure it adheres to it

AkA84 commented:
Done, I also checked the IA with the browsers and devices that I have available at the moment

zekiel commented:
many thanks!

phalt commented:
Hi, PokéAPI author here. I'm just commenting to say you've made my day :joy:

phalt commented:
@zekiel concerned about extra traffic hitting the servers - do you cache these queries at all? I should be able to handle quite a bit of traffic (already on 30k+ a day)

moollaza commented:
Hey @phalt!

>I'm just commenting to say you've made my day

Haha I'm so glad to hear that! I didn't realize you were also responsible for swapi.co which I believe someone else is planning on integrating into an Instant Answer :+1: 

>do you cache these queries at all? I should be able to handle quite a bit of traffic (already on 30k+ a day)

Yes, absolutely! We can cache this for as long as you'd like. Seeing as this stuff is pretty static we can increase the caching as much as you'd like. 1 week? 1 month? 

@AkA84 All that needs to be done is to set `spice proxy_cache_valid` to something like `200 1m`. This would cache HTTP 200 responses for 1 months.

More details: https://duck.co/duckduckhack/spice_advanced_backend#caching

phalt commented:
@moollaza Any reasonable caching is fine. As I said - the tough little server I have this running on can take upwards of 30k hits a day with no problems, so scaling it a bit more shouldn't be too hard. If it get bad, who can I email?

moollaza commented:
>scaling it a bit more shouldn't be too hard. If it get bad, who can I email?

@phalt sounds good! If you need anything, feel free to email me: moollaza@duckduckgo.com

AkA84 commented:
> @AkA84 All that needs to be done is to set spice proxy_cache_valid to something like 200 30d. This would cache HTTP 200 responses for 1 months.

Done!

> Hi, PokéAPI author here. I'm just commenting to say you've made my day :joy:

@phalt I'm really happy to hear that! Since you're here :smile: a couple of things:
* Just FYI: the Instant Answer will make two calls to the API, one to the *pokemon/{name}* endpoint and a second one to */description/{id}* to get the pokemon's description
* Speaking about additional calls: I wanted to avoid making a separate call to get the pokemon's sprite (via the */sprite/{id}*), so I'm fetching it directly on */media/img/{national_id}.png*. Do you see any potential problem with that? So far with all the tests I've made it worked pretty good

phalt commented:
@AkA84 
- Perfectly fine and that's how it is designed - RESTful!
- No problem doing that at all, it's just a direct link to the image file.

kasarov55555 commented:
I am so glad that this is being added to Duckduckgo, but in the example(pokemon pikachu, bulbasaur pokemon) you have to add the word Pokemon. Is there any way that the word Pokemon could be made not mandatory?(all you have to do is search "pikachu" or "buizel")

AkA84 commented:
@kasarov55555 that is something that aimed at when I started working on the IA, but I wouldn't know how it worked if you didn't provide a static "hook" for the IA to be triggered with. 

At the same time I see that there is for example the [Definition IA](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/Dictionary/Definition.pm) that although it lists a series of triggers, it still shows up if you simply write a word (for example it works with both *"awesome definition"* and *"awesome"*)

I'd really like the IA to be triggered by queries like *"pikachu"* or *"pikachu stats"*, but being the first time I develop one I'd need the help from the DDG guys :)

MrChrisW commented:
@AkA84 I've been following this PR and it looks awesome! Great to see the API owner is here too :+1: 

>I'd really like the IA to be triggered by queries like "pikachu" or "pikachu stats", but being the first time I develop one I'd need the help from the DDG guys :)

You could get this IA triggering on just a pokemon name "pikachu" however that would require maintaining a list of all known names and slurping that for use as triggers. An example is the [Airlines IA](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/Airlines.pm#L25) which uses a list of known Airline in [a text file](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/share/spice/airlines/airlines.txt)

The only issues I see with doing this is the maintainability, e.g if more pokemon are created the list needs updating. 

On another note,
I've also noticed there seems to be an internal IA using the "Pokemon Encyclopedia" as a source which is triggered by name https://duckduckgo.com/?q=Buizel&ia=about 
There's also the standard "Wikipedia" IA which triggers as well https://duckduckgo.com/Pikachu?ia=about although these can potentially be changed (I think) if a better option is available :smile:

AkA84 commented:
> @AkA84 I've been following this PR and it looks awesome!

@MrChrisW thanks!

> You could get this IA triggering on just a pokemon name "pikachu" however that would require maintaining a list of all known names and slurping that for use as triggers. 

As you said I don't think it's very maintainable and the list of pokemon is quite long! Also, if you need to manually amend a list, where's the fun in that :smile: 

I'm still curious about how the Dictionary IA works, because it seems to be triggered even without any static trigger in the query!

 > I've also noticed there seems to be an internal IA using the "Pokemon Encyclopedia" as a source which is triggered by name https://duckduckgo.com/?q=Buizel&ia=about 

Mmm this is not good, I see a bit of overlapping here :-/

MrChrisW commented:
>As you said I don't think it's very maintainable and the list of pokemon is quite long! Also, if you need to manually amend a list, where's the fun in that 

:smile: Exactly! Best to let the API handle the updating :wink: 

>I'm still curious about how the Dictionary IA works, because it seems to be triggered even without any static trigger in the query!

I'm guessing that's due to internal triggers configured by DDG, specificity these triggers may detect if a dictionary link is returned in the SERP and then the Dictionary IA is called. (I'm just guessing) 

>Mmm this is not good, I see a bit of overlapping here :-/

Not to worry, I'm sure @moollaza will know what to do! :smiley:

phalt commented:
@MrChrisW @AkA84 Just call the `/pokedex/` endpoint to get a list of all Pokémon names, you won't need to hardcode the list that way :)

There are other Pokémon triggers - I'd suggest keeping `Pokemon <NAME>` for now so there isn't any confusion with these.

jagtalon commented on lib/DDG/Spice/Pokemon/Data.pm:
```diff
@@ -0,0 +1,31 @@
+# Triggers
+triggers startend => "pokemon";
+
+spice to => 'http://pokeapi.co/api/v1/pokemon/$1';
```
Please add a `/` at the end of `$1` because http://pokeapi.co/api/v1/pokemon/pikachu redirects to it. We can't handle redirects in production machines unfortunately.

AkA84 replied:
Done!

moollaza commented on share/spice/pokemon/data/pokemon_data.js:
```diff
@@ -0,0 +1,119 @@
+            }
+        });
+
+        if( api_result.descriptions.length > 0 ) {
```
@AkA84 would you mind moving all these private function definitions outside the `env.ddg_spice_pokemon_data` function definition?

AkA84 replied:
Done! I put them there to more easily convey at a glance the fact that they're only used by the `Pokemon::Data` endpoint and not also by the `Pokemon::Description` one, but this way `env.ddg_spice_pokemon_data` looks way more readable :+1:

moollaza replied:
@AkA84 thanks a lot! I definitely understand your reasoning, but I have a suggestion, so that we can remove the `Pokemon::Description` callback! Will comment on that line...

moollaza commented on share/spice/pokemon/data/pokemon_data.js:
```diff
@@ -0,0 +1,118 @@
+    /**
+     *  [ Pokemon::Description ]
+     */
+    env.ddg_spice_pokemon_description = function(api_result){
```
@AkA84 I have a suggestion for this. We can avoid relying on a separate callback function, if we simply use `$.getJSON` (you'll need to remove the callback from the `spice to` URL).

We can also call our `$.getJSON` using the Spice `onShow` [event](https://duck.co/duckduckhack/spice_displaying#events).

This way when the IA displays, we'll react, load the JSON, and then put the content in the right place.

This way everything will only rely on a single callback, `ddg_spice_pokemon_data`

moollaza replied:
The ISBN Spice is a great example of this: https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/share/spice/isbn/isbn.js#L31

AkA84 replied:
@moollaza thanks for the suggestion, it's really a nice improvement! Two things though:

1. When do you need multiple callbacks then? For example I see that that approach is used for the [Dictionary AI](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/share/spice/dictionary/definition/dictionary_definition.js)
2. I ended up using the `onItemShown` event handler, but in your [wiki](https://duck.co/duckduckhack/spice_displaying#events) you don't have it listed. Can I keep it like this or should I use the `onShow` one?

moollaza replied:
@AkA84 great question. Originally we didn't have jQuery at our disposal so we relied on multiple callbacks. We'd use a function to create a script tag on the page which would cause the additional callback(s) to fire.

Now that we have jQuery, `$.getJSON` is simpler to use IMO and we now have the events which means a lot of the older IA just need to be updated and simplified.

moollaza replied:
In a few rare cases, we actually rely on the 2nd API call to provide the primary info (i.e. the first endpoint gives us a search result, the 2nd api call gets us the data on the search result) and so that's one case where we treat things differently in the the `$.loadJSON()` callback function actually fires the `Spice.ad()` call, because we need to wait for that data to display our result.

moollaza replied:
To answer #2, you'll want to use `onShow`. `onItemShown` get fired as each item (tile) is rendered. You would use that to modify each tile as they display, which is how we display the ratings for Amazon tiles. However, `onShow` fires once when the tab opens which is what we want to pay attention to.

AkA84 replied:
> To answer #2, you'll want to use `onShow`

Done!

AkA84 commented:
@moollaza is there anything else that I need to improve? If not, could you help me out with this

> On mobile the "auxiliary context" box disappears for my IA, but not for others. I noticed that the .cw wrapper is supposed to also get an .has-aux class to prevent it, but I'm not sure how to add that

This causes the IA to be rendered not in an optimal way on mobile, so I'd really like to fix it

moollaza commented:
@AkA84 at what width is it disappearing? It's expected that we don't show the aux both for any IA's on Mobile because there simply isn't enough space to display it. We're going to be revisiting and re-doing the auxiliary box soon though and I believe the plans include a way to expose it on mobile.

AkA84 commented:
@moollaza thanks for your quick reply! The breakpoint is at 959px. But if you search for, for example, *"New York Times"*, you will see that in the *Info* IA (which seems to be using the standard `basic_info_detail` template) although the aux box disappears if you load the page in a desktop-like width and then resize the viewport, if you load the page in a mobile/desktop width you will get a nice *"Show more"* button that will display all the data!

If I manually add to my IA via inspector the class `.has-aux` to the `.cw` wrapper I can see that the aux box is displayed (still missing the *"Show more"* button of course) so I was wondering how I could configure the IA so that it adds it automatically. Or maybe it's something that only the built-in IAs like *Info* have?

moollaza commented:
@AkA84 Ahhh I see what you mean now. That seems like a bug to me! I'll let the team know and we should be able to move forward with this IA.

My only other thought is that you should remove the italics from the "description". We try to avoid using it unless its really necessary. Once that's fixed I'll put it up on a testing machine.

-----

Ideally I would prefer you use the `record` template to produce your table content, but I see you're adding a link to the table which you currently can't do if you use our record template to generate the data.

You could move the link to the aux box, but then there won't be much to show in the description section. 

Also we have a `subtitle` property now for the `info` template, which I think would be best for the current "description" text, but I noticed the subtitle looks a little odd when its long as we have a padding issue. I've notified the Engineering team to fix that.

AkA84 commented:
> My only other thought is that you should remove the italics from the "description". We try to avoid using it unless its really necessary. Once that's fixed I'll put it up on a testing machine for a design signoff

Done, I replace the italic with quotes

> Ideally I would prefer you use the record template to produce your table content, but I see you're adding a link to the table which you currently can't do if you use our record template to generate the data.
>
> You could move the link to the aux box, but then there won't be much to show in the description section. 

Exactly my train of thought :smile: 

> Also we have a subtitle property now for the info template, which I think would be best for the current "description" text, but I noticed the subtitle looks a little odd when its long as we have a padding issue. I've notified the Engineering team to fix that.

I wouldn't know how to make it work though, as the description is fetched asynchronously. Could I still use the property somehow?

moollaza commented:
@AkA84 I can think of a few ways for you can get the subtitle to work:

1. Wait for the description JSON response, then simply navigate the DOM, find the header and insert a new tag after it, with the correct class and content (`<div class="zci__subheader">...</div>`)
2. Provide placeholder text for the subtitle (I'm not sure if a string of spaces would work?) and set the subtitle to `visibility:hidden` in the CSS. Then once the description text is received, replace the content as required and force the subtitle to display
3. The third option is to move the `Spice.display()` call into the callback function for your description's `$.GetJSON()` -- this of course delays the rendering slightly, but means you don't have to bother messing with the DOM. --- I'm kinda leaning towards this as the simplest solution.

Thoughts?

AkA84 commented:
@moollaza right, I was thinking about how to use the subtitle strictly in the context of the `onShow` callback!

If I have to choose I personally prefer the second option as I would like the user to see the IA as soon as possible since what he/she is most likely interested in are the stats, types, etc. The description is mostly there for "decoration" so I think it's not a problem if it shows up an instant later (unless you guys don't like flash of appended content in the IAs)

To make things simpler without pushing another commit, the solution would look like this:

![example](https://cloud.githubusercontent.com/assets/6400898/6469899/6bc81a7e-c1e0-11e4-9412-228788cd0201.png)

```javascript
normalize: function(item) {
    return {
        title: item.name,
        subtitle: ' ', // placeholder, to be filled with the description

// ....

onShow: function() {
    // ...

    Spice.getDOM(ID).find('.zci__header__sub').html(description).addClass('is-visible');
```
```css
.zci--pokemon .zci__header__sub {
    quotes: "“" "”";
    visibility: hidden;
}
.zci--pokemon .zci__header__sub.is-visible {
    visibility: visible;
}
.zci--pokemon .zci__header__sub:after {
    content: close-quote;
}
.zci--pokemon .zci__header__sub:before {
    content: open-quote;
}
```

---

About the CSS issues with the subtitle, moving the padding to the title itself would solve the problem, so that it doesn't matter if the subtitle goes on on or two lines:

```css
.zci__header.has-sub > a {
    padding-right: 0.5em;
}

.zci__header__sub {
  /* padding-left: 0.5em;  <= removed */
  display: inline-block;
  text-transform: capitalize;
}
```
I'm not sure why the `.zci__header__sub` has the `capitalize` applied though, I think it kinda looks weird with long texts like a pokemon's description

---

Let me know what you think, thanks!

moollaza commented:
@AkA84 I would remove the quotes as we don't really do that anywhere else and add CSS to prevent the the capitalization from being applied.

@abeyang any thoughts?

moollaza commented:
Available for testing: https://ddh8.duckduckgo.com/?q=pokemon+bulbasaur&ia=pokedex

AkA84 commented:
> Available for testing: https://ddh8.duckduckgo.com/?q=pokemon+bulbasaur&ia=pokedex

I'm happy already just to see it live here :smile:

@moollaza you think I should already push a commit to remove the quotes and use `subtitle` (even with the padding problem), or should i wait for the design review first?

moollaza commented:
@AkA84 Yes, please go ahead and remove the quotes. You can leave the rest for now.

moollaza commented:
@AkA84 spoke with the design team, for now lets simplify things and move the three stats, Type, Egg Group and Evolves into to the Aux box on the right. We could put it under a new heading at the top so it's still prominently displayed.

One thought I just had, maybe the type should be the Subtitle? That seems reasonable to me?

Then we can keep the description text as the body of the IA, which would greatly simplify the template as well.

Would look something like this...
![2015-03-03_17h55_32](https://cloud.githubusercontent.com/assets/873785/6474412/7eab4dee-c1ce-11e4-943b-d7c4052d9030.png)

/cc @abeyang

moollaza commented:
Also, @AkA84 or @phalt are there any large images available? Or might there be in the future?

AkA84 commented:
@moollaza ok I followed your suggestions and now it definitely looks better!

![new-design](https://cloud.githubusercontent.com/assets/6400898/6491150/9095bd68-c2ab-11e4-8f1e-e3c7c481bf87.png)

Things to note:
* I suppose you're using your Handlebars `concat` helper in `subtitle` and in the aux box when being provided with an array, since you're concatenating the elements with a `,`. Problem is (or maybe it's intentional, not sure) that there is no space after the comma and it doesn't look as good as it could
* I increased the max length of the description since now there's more room for it
* I had to add a bit of `margin-top` to `.info--head` to separate **Stats** from *Egg groups* and *Evolves into*, and because of IE8 and its lack of support for `:not()`, I had to resort to this verbose thing here in order not to apply the margin where the stats are the only thing available (it can happen for the last evolution of a pokemon):
```css
.zci--pokemon .tile--info .info--head {
    margin-top: 10px;
}
.zci--pokemon .tile--info .info--head:first-child {
    margin-top: 0;     
}
```

---

> Also, @AkA84 or @phalt are there any large images available? Or might there be in the future?

As far as I know there aren't large images available, but I'd say @phalt is in a better position to answer your question :smile:

jagtalon commented:
@AkA84 awesome--testing it!

jagtalon commented:
Nice. It works all the way down to IE 8. @abeyang @moollaza https://ddh9.duckduckgo.com/?q=pikachu+pokemon&ia=pokedex

moollaza commented:
>I suppose you're using your Handlebars concat helper in subtitle and in the aux box when being provided with an array, since you're concatenating the elements with a ,. Problem is (or maybe it's intentional, not sure) that there is no space after the comma and it doesn't look as good as it could

@AkA84 actually we are expecting `subtitle` to be a string we're not running any functions on it, so potentially Handlebars is handling it in some clever way to produce the comma-separated values. Can you update your code to return a properly formatted string?

> I increased the max length of the description since now there's more room for it

Oh, I didn't notice you were doing that. There's no need! We automatically chomp the text and display a "Show more" button that will expand the IA to show all the text and auto-opens the info box :)


>I had to add a bit of margin-top to .info--head to separate Stats from Egg groups and Evolves into, and because of IE8 and its lack of support for :not(), I had to resort to this verbose thing here in order not to apply the margin where the stats are the only thing available (it can happen for the last evolution of a pokemon):

Hmm, I actually just modified the DOM in place to produce the picture I posted above. I was hoping our Infobox/CSS would cleverly handle the case of a title after an item. Your CSS looks OK. Alternatively we could remove the "Stats" heading.

AkA84 commented:
> Can you update your code to return a properly formatted string?
> Oh, I didn't notice you were doing that. There's no need!
> Actually I just realized you can set url, label and value to achieve basically what you're doing here. 

All done!

phalt commented:
Hi folks, the images from the API are the only images (They're pretty small)

moollaza commented:
@AkA84 excellent!

I noticed one more thing, when we have no description (and picture) the result looks a little broken. Let's put "Description not available" if we don't have one:

![2015-03-05_11h59_59](https://cloud.githubusercontent.com/assets/873785/6509702/27d37da4-c32f-11e4-822d-9df51f631c86.png)

AkA84 commented:
@moollaza done!

abeyang commented:
Hi @AkA84, think we're almost there! Can we move the stats to the top of the description pane, similar to "card details" in the [hearthstone IA](https://github.com/duckduckgo/zeroclickinfo-spice/pull/1452#issuecomment-75341127)? If "stats" is not the right word, then maybe we can try "**Details**" or "**Info**"?

AkA84 commented:
Hi @abeyang ! I'm not sure what you mean, you want me to put *Egg groups* and *Evolves into* under the **Stats** heading? Or you want to have first the Stats and then those two info?

abeyang commented:
@AkA84 I'm fine with the ordering; would just want the heading at the very top.

AkA84 commented:
@abeyang done :)
@moollaza I removed the custom CSS for the infobox heading since at this point it wasn't needed anymore

abeyang commented:
@AkA84 , it's getting close, but noticed a few more things. I just realized that you're getting a 120x120 mostly translucent png for the images, which is causing it to look imbalanced from the rest of the page. Can you add a white backing and gray border, so it looks more like what you see on the right column here:

![screen shot 2015-03-06 at 2 58 49 pm](https://cloud.githubusercontent.com/assets/794084/6538035/936f7b7a-c411-11e4-818c-12358ddee1ed.png)

Finally, I noticed that the "evolves into" is clickable. I don't like how it breaks the right-alignment of the text, though I think you did that to highlight the fact that it is a link. So here's my proposal: what if we move that from the right pane and put it directly into the subtitle, so that for Pikachu, it would say:

> Electric | Evolves into Raichu

It's similar to what we have for [Twitter](https://duckduckgo.com/?q=%40yegg&ia=social), so you can borrow elements from that.

LMK if you have any questions, thanks!

AkA84 commented:
@abeyang 

> Can you add a white backing and gray border

Done. I've also added a bit of padding so that the image has a bit of room and doesn't end up too close to the border

> I don't like how it breaks the right-alignment of the text, though I think you did that to highlight the fact that it is a link.

Well actually this is how it's shown by default by the infobox, it's just how the DDG built-in template works, so I assume that it's implemented like this in another IA at least?

I would like to keep the *Evolves into* line in the infobox if possible as I think it belongs there at this point (for now I moved it at the top of the list so the misalignment doesn't happen in the middle of it). One possible solution would be to make visible by default the little arrow that you get when hovering over the link, and that would solve the alignment problem and also it would make more obvious to the user that that's actually a link. But I'd need also @moollaza approval for that I think since it's overriding the default behaviour. Something like this:

![new-take](https://cloud.githubusercontent.com/assets/6400898/6538465/d82a2a70-c462-11e4-92a7-5142f5aeeafc.png)

What do you think? Thanks!

abeyang commented:
@AkA84 unfortunately, that would break our conventions. Currently, for our info pane, we have IAs that:

- [only show text](https://duckduckgo.com/?q=facebook&ia=about)
- [only show links](https://duckduckgo.com/?q=julius+caesar&ia=about)

--but never both. But here's another suggestion: what if we put the *type* inside the info pane, and move the evolution grouping into the byline? So, in your example, it might look like:

![screen shot 2015-03-06 at 4 03 01 pm](https://cloud.githubusercontent.com/assets/794084/6538697/5043ced8-c41a-11e4-8993-74f850b8ef56.png)

What do you think?

PS: Oh, one more thing -- I think we can disable the self-linking in the title.

AkA84 commented:
@abeyang ok i moved things around as suggested!

One thing though about the image: I added a bit of padding around it because of the background colour, but now it doesn't look good on mobile as on smaller screens the image has a fixed height

![screen](https://cloud.githubusercontent.com/assets/6400898/6539319/7b809ebc-c471-11e4-804b-b1da87d1e906.png)

I wanted to take a look at how things are done in the IAs you put screenshot of before but with none of them I managed to get the result on the right column, I just see the image full-width. Could you help me with this?

> PS: Oh, one more thing -- I think we can disable the self-linking in the title.

I'm not sure how to do this, the link is applied automatically it seems

moollaza commented:
@AkA84 I just found out that you can automatically produce the white background/border for an image if you set `imageIsLogo: true,` in your normalize function.

It also seems to handle mobile images better:

![2015-03-10_15h53_00](https://cloud.githubusercontent.com/assets/873785/6584437/a2df99d6-c741-11e4-8a34-fd842959bbaa.png)

AkA84 commented:
@moollaza oh that's great, I'll try it out as soon as I can thanks! Was this documented somewhere?

moollaza commented:
@AkA84 sadly not -- I wasn't even aware of that one! Don't worry though, we're going to be doing a big overhaul of the docs. So soon nothing will be a mystery any more!

AkA84 commented:
@moollaza @abeyang done, now the image looks good :)

abeyang commented:
Looks good! /cc @moollaza

phalt commented:
Oh my GOSH does this work now? :)

MrChrisW commented:
@phalt Not just yet! A new version of the Spice repositories will need to be released and deployed on duckduckgo. @moollaza Will provide a link and update here once it goes live!! :+1:

phalt commented:
@MrChrisW \o/ this is the coolest thing ever that's happened to one of my projects. @AkA84 thank you so much! I owe you a :beer:

AkA84 commented:
@phalt apparently it's live https://duckduckgo.com/?q=pikachu+pokemon&ia=pokedex ! :smile:

phalt commented:
@AkA84 Woo!

moollaza commented:
@AkA84 you beat me to it! It went live today :+1: 

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's & Issues or even submit more Instant Answers!

phalt commented:
I've seen an uptick in hits (now around 100k a day, was at around 40k) and the server is handling it just fine :)

moollaza commented:
@phalt that's awesome -- I'm wondering if we're severely over triggering now though. I just realized we're not checking to see if the remainder is only a single word -- we should probably do that so we don't send you API request for irrelevant queries like "pokemon movie release date"

All Pokemon names are a single word right? i.e. if the remainder has a space in it, we can bail?

/cc @AkA84

phalt commented:
@moollaza Yes, no spaces. Single word.

moollaza commented:
@AkA84 would you be interested in adding that^ fix?

AkA84 commented:
@moollaza #1693 :)

moollaza commented:
@AkA84 awesome thanks for making that! @phalt I'll be releasing this evening, should see the changes go live within the next few days.

AkA84 commented:
@moollaza any news on when the IA will be back live? Is there anything i can do on my end?

moollaza commented:
Sorry for the delay. It will be live again with the next deploy, that should be some time this week!
